packages/office-addin-dev-settings/src/appcontainer.ts

```diff
@@ -143,7 +143,7 @@ export async function getAppcontainerNameFromManifest(manifestPath: string): Pro
   const sourceLocation = manifest.defaultSettings ? manifest.defaultSettings.sourceLocation : undefined;
 
   if (sourceLocation === undefined) {
-    throw new Error(`The source location could not be retrieved from the manifest.`);
+    throw new ExpectedError(`The source location could not be retrieved from the manifest.`);
```

Comment:
How are we defining what is an expected error and what isn't?

Reply:
I think there is some subjectivity in this.
But the idea is that ExpectedError = errors that are expected to happen given a specific flow of actions

Reply:
Unexpected errors are things where the code itself has a problem, e.g. our code crashes. If the outcome is because of the inputs from the user, then it is an expected error. We cannot do anything to resolve the error. When we calculate reliability, we can know the overall success/total (ICE %), and we can know the unexpected errors / total (ACE%). We want our ACE% to be 100% ideally, e.g. our code has no issues, and we can monitor ICE% to see if there areas where customers are running into problems preventing them from being successful.

Reply:
It's not really subjective. An error which is the result of the inputs from the user are expected errors. We cannot change our code to influence the outcome. An unexpected error is a crash in our code.

Reply:
The reason I asked is because I have seen differing points of view on this in recent months.  Martyn seems to prefer a stricter approach where there aren't many cases where things are "expected" (the context in that discussion was auth and things like needing to refresh from a token was not considered expected).

In this case I see that the defaultSettings were undefined and so the source Location can be retrieved, but that doesn't explain why it's a problem the user created.

Reply:
I see. I think it would be better to have a separate check to throw an Error if manifest.defaultSettings is not defined.

Reply:
Actually, I take that back. Both of these are just based on the manifest being correct. If the manifest is missing a DefaultSettings section or a SourceLocation field, that would cause the error, which is ExpectedError because it is based on user input.

Reply:
Could it be a result of errors in the manifest parsing code rather then the manifest itself?  Is the manifest malformed because of something we did or the customer explicitly did?  These may be somethings we really can't determine . . . I just want us to be careful about our assumptions on what is expected vs. not expected.

Reply:
Not likely.